src/FlexVotingClient.sol

```diff
@@ -0,0 +1,169 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.10;
+
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {Checkpoints} from "@openzeppelin/contracts/utils/Checkpoints.sol";
+import {IFractionalGovernor} from "src/interfaces/IFractionalGovernor.sol";
+import {IVotingToken} from "src/interfaces/IVotingToken.sol";
+
+// TODO natspec
+// this is a contract to make it easy to build clients for flex voting governors.
+abstract contract FlexVotingClient {
+  using SafeCast for uint256;
+  using Checkpoints for Checkpoints.History;
+
+  /// @notice The voting options corresponding to those used in the Governor.
+  enum VoteType {
+    Against,
+    For,
+    Abstain
+  }
+
+  /// @notice Data structure to store vote preferences expressed by depositors.
+  struct ProposalVote {
+    uint128 againstVotes;
+    uint128 forVotes;
+    uint128 abstainVotes;
+  }
+
+  /// @notice Map proposalId to an address to whether they have voted on this proposal.
```

Comment:
we usually use `/// @notice` is for user-facing/public/external, and `/// @dev` for internal/private stuff

Reply:
Fixed in 1ac8bc2, thanks